src/pcapi/scripts/payment/batch_steps.py

```diff
@@ -1,86 +1,128 @@
 from pcapi.domain.reimbursement import RULES
 from pcapi.domain.reimbursement import find_all_booking_reimbursements
 from pcapi.models.db import db
 from pcapi.models.payment import Payment
+from pcapi.models.payment_message import PaymentMessage
 from pcapi.models.payment_status import TransactionStatus
 from pcapi.repository import payment_queries
 from pcapi.repository import repository
 from pcapi.repository.user_queries import get_all_users_wallet_balances
-
-
-logger = logging.getLogger(__name__)
 from pcapi.utils.mailing import MailServiceException
 
 
-def concatenate_payments_with_errors_and_retries(payments: list[Payment]) -> list[Payment]:
-    error_payments = payment_queries.find_error_payments()
-    retry_payments = payment_queries.find_retry_payments()
-    payments = payments + error_payments + retry_payments
-
-    logger.info("[BATCH][PAYMENTS] %i Payments in status ERROR to send", len(error_payments))
-    logger.info("[BATCH][PAYMENTS] %i Payments in status RETRY to send", len(retry_payments))
-    logger.info("[BATCH][PAYMENTS] %i Payments in total to send", len(payments))
-
-    return payments
-
-
-def generate_new_payments() -> tuple[list[Payment], list[Payment]]:
-    offerers = Offerer.query.all()
-    all_payments = []
+logger = logging.getLogger(__name__)
 
-    for offerer in offerers:
-        booking_reimbursements = []
-        for venue in offerer.managedVenues:
-            final_bookings = booking_repository.find_bookings_eligible_for_payment_for_venue(venue.id)
-            booking_reimbursements += find_all_booking_reimbursements(final_bookings, RULES)
 
-        booking_reimbursements_to_pay = filter_out_already_paid_for_bookings(
-            filter_out_bookings_without_cost(booking_reimbursements)
+def include_error_and_retry_payments_in_batch(batch_date: datetime):
+    statuses = [TransactionStatus.RETRY, TransactionStatus.ERROR]
+    payments = payment_queries.get_payments_by_status(statuses)
+    # We cannot directly call "update()" when "join()" has been called.
+    ids = payments.with_entities(Payment.id)
+    payments = db.session.query(Payment).filter(Payment.id.in_(ids))
+    payments.update({"batchDate": batch_date}, synchronize_session=False)
+    db.session.commit()
+
+
+def get_venues_to_reimburse() -> Iterable[tuple[id, str]]:
+    return tuple(
+        db.session.execute(
+            """
+        SELECT id, CASE WHEN "publicName" IS NOT NULL THEN "publicName" ELSE name END
+        FROM venue
+        WHERE id IN (
+            SELECT DISTINCT(venue.id)
+            FROM venue
+            JOIN offer ON offer."venueId" = venue.id
+            JOIN stock ON stock."offerId" = offer.id
+            JOIN booking ON booking."stockId" = stock.id
+            LEFT OUTER JOIN payment ON payment."bookingId" = booking.id
+            WHERE
+                booking."isUsed" AND NOT booking."isCancelled"
+                AND payment.id IS NULL
         )
+            """
+        )
+    )
```

Comment:
```suggestion
booking_alias = aliased(Booking)
    venues = (
        Venue.query.join(Offer)
        .join(Stock)
        .join(booking_alias)
        .outerjoin(Payment, and_(booking_alias.id == Payment.bookingId, Payment.id is None))
        .filter(and_(booking_alias.isUsed, not booking_alias.isCancelled))
        .filter(Payment.id is None)
        .with_entities(Venue.id, Venue.publicName, Venue.name)
        .all()
    )
    result = []
    for venue_tuple in venues:
        result.append((venue_tuple.id, venue_tuple.publicName if venue_tuple.publicName else venue_tuple.name))
    return result
```

Ça te paraît jouable ? :)

Reply:
J'ai adapté un tout petit poil, mais ça marche. Merci !